### PR TITLE
New page on how to upload a large binary file

### PR DESCRIPTION
**Pull request by ammaboateng**
A step by step guide on how to upload large binary files using the `garage-sign` tool.

Relates-to: OTA-5002

Signed-off-by: amma <awuraaamma.okyere-boateng@here.com>

### CODE REVIEW

File: docs/ota-client-guide/modules/ROOT/nav.adoc [hunk 1]
```diff
@@ -85,6 +85,7 @@ ifndef::env-github[:pageroot:]
 * xref:{pageroot}build-only-ostree.adoc[Build only the OSTree part]
 * xref:{pageroot}rollback.adoc[Set up rollback behavior]
 * xref:{pageroot}deb-package-install.adoc[Install the client from a deb package]
+* xref:{pageroot}upload-large-binary.adoc[Upload a Large Binary file]
```
Comment (patrickvacek) [inline docs/ota-client-guide/modules/ROOT/nav.adoc:88]: Typically we follow sentence capitalization except for proper names, which I don't think applies to "Large Binary". :)

File: docs/ota-client-guide/modules/ROOT/pages/upload-large-binary.adoc [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+----
+garage-sign targets add-uploaded \
+  --repo <reponame> \
+  --input </path/to/bigfile.bin> \
+  --name <name-of-target> \
+  --version <target-version> \
+  --hardwareids <hardware-id1>,<hardware-id2> <1>
+----
++
+<1> This refers to the ids of hardware that your binary file supports. An example can be seen in the `targets.json` file xref:customise-targets-metadata.adoc#_anatomy_of_targets_json_metadata[here] under the `custom` field.
```
Comment (patrickvacek) [inline docs/ota-client-guide/modules/ROOT/pages/upload-large-binary.adoc:50]: Perhaps more helpful:

```suggestion
<1> This refers to the types of hardware that your binary file is intended for. An example can be seen in the `targets.json` file xref:customise-targets-metadata.adoc#_anatomy_of_targets_json_metadata[here] under the `custom` field.

```

File: docs/ota-client-guide/modules/ROOT/nav.adoc (hunk 1 above)
Comment (ammaboateng) [reply docs/ota-client-guide/modules/ROOT/nav.adoc:88]: @patrickvacek good point. I'll make the change.

Comment (patrickvacek) [reply docs/ota-client-guide/modules/ROOT/nav.adoc:88]: The reference here in the nav is still capitalized inconsistently.

Comment (ammaboateng) [reply docs/ota-client-guide/modules/ROOT/nav.adoc:88]: @patrickvacek it's strange that my changes weren't committed. I'll try it again.

Comment (ammaboateng) [inline docs/ota-client-guide/modules/ROOT/nav.adoc:88]:
```suggestion
* xref:{pageroot}upload-large-binary.adoc[Upload a large binary file]
```

File: docs/ota-client-guide/modules/ROOT/pages/upload-large-binary.adoc [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+====
+endif::[]
+
+Software packages can be uploaded on the OTA Connect Portal by following the section xref:ota-web::upload-software-ui.adoc[Upload software versions, window="_blank"] of the OTA Connect User Guide.
+
+You can also upload large binary files (up to 3 GB) using the `garage-sign` tool on the command line.
+
+*To upload a large binary file, follow these steps:*
+
+. Ensure you have the latest version of the `targets.json` file from OTA Connect by pulling the latest version to your local repository.
```
Comment (xaviergully) [inline docs/ota-client-guide/modules/ROOT/pages/upload-large-binary.adoc:16]: Do we assume that users already have a local repo server ? 
Or shall we had an initialization step, i.e garage-sign init ?

Comment (ammaboateng) [reply docs/ota-client-guide/modules/ROOT/pages/upload-large-binary.adoc:16]: @xaviergully good point. I will also add the initialization step.

File: docs/ota-client-guide/modules/ROOT/pages/upload-large-binary.adoc [hunk 4]
```diff
@@ -0,0 +1,79 @@
+= Upload a binary file
+ifdef::env-github[]
+
+[NOTE]
+====
+We recommend that you link:https://docs.ota.here.com/ota-client/latest/{docname}.html[view this article in our documentation portal]. Not all of our articles render correctly in GitHub.
+====
+endif::[]
+
+If you want to upload large binary files (up to 3 GB) or your software signing keys are xref:ota-client::rotating-signing-keys.adoc[taken offline], use the `garage-sign` tool on the command line.
```
Comment (patrickvacek) [inline docs/ota-client-guide/modules/ROOT/pages/upload-large-binary.adoc:10]:
```suggestion
If you want to upload large binary files (up to 3 GB) or your software signing keys are xref:ota-client::rotating-signing-keys.adoc[offline], use the `garage-sign` tool on the command line.
```

Alternatively, "have been taken offline" works too, but it's more verbose.

File: docs/ota-client-guide/modules/ROOT/pages/upload-large-binary.adoc [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+NOTE: You can upload software packages of up to 1 GB on the OTA Connect Portal. For instructions, see xref:ota-web::upload-software-ui.adoc[Upload software versions] in the OTA Connect User Guide.
+
+*To upload a binary file using `garage-sign`:*
+
+. Make sure you have the link:https://tuf-cli-releases.ota.here.com/index.html[latest version,window="_blank"] of the `garage-sign` tool.
+. Get the .zip file with your provisioning credentials.
++
+For instructions, see the xref:ota-client::generating-provisioning-credentials.adoc[related] section in this guide.
+
+. To initialize a local repository, specify its name and path to the .zip file with your provisioning credentials.
```
Comment (patrickvacek) [inline docs/ota-client-guide/modules/ROOT/pages/upload-large-binary.adoc:21]:
```suggestion
. To initialize a local repository, specify a name and the path to the .zip file with your provisioning credentials.
```

**Comment by xaviergully** [comment]
@HalynaDumych , one small comment, could you make a reference to the jira ticket OTA-5002, this commit is resolving ?

File: docs/ota-client-guide/modules/ROOT/pages/upload-large-binary.adoc (hunk 4 above)
Comment (HalynaDumych) [reply docs/ota-client-guide/modules/ROOT/pages/upload-large-binary.adoc]: Changed

File: docs/ota-client-guide/modules/ROOT/pages/upload-large-binary.adoc (hunk 5 above)
Comment (HalynaDumych) [reply docs/ota-client-guide/modules/ROOT/pages/upload-large-binary.adoc]: Changed

**Comment by HalynaDumych** [comment]
> I thought we agreed to handle this as part of the overal garage-sign reference documentation. I'm not a fan of spending time on making an individual page for this right now; we're just going to have to refactor it when the rest of the garage-sign documentation is ready.

Hm, it's what was specified in the current task: https://saeljira.it.here.com/browse/OTA-5002

Should all garage-sign reference documentation appear on one page? Or just in one section?

**Pull request merged**